react/ContactsList/ContactsList.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+
+import { sortLastNameFirst, buildLastNameFirst } from './'
+import ContactRow from './ContactRow'
+import ContactHeaderRow from './ContactHeaderRow'
+import styles from './styles.styl'
+
+const ContactsList = props => {
+  const { contacts, className, onItemClick, ...rest } = props
+
+  const sortedContacts = [...contacts].sort(sortLastNameFirst)
+  const categorizedContacts = sortedContacts.reduce((acc, contact) => {
+    const name = buildLastNameFirst(contact)
+    const header = name[0] || 'EMPTY'
+    acc[header] = acc[header] || []
+    acc[header].push(contact)
+    return acc
+  }, {})
+
+  return (
+    <ol className={cx(styles['list-contact'], className)} {...rest}>
+      {Object.keys(categorizedContacts).map(header => (
```

Comment:
Nit: It is not guaranteed here than the keys will be returned in the order of insertion. I think sorting the keys would be better to ensure that we have no surprise.